htdocs/js/jsx/behavioural_feedback_panel.jsx

```diff
@@ -0,0 +1,606 @@
+
+    var that = this;
+    var feedbackID = this.props.feedbackID
+    var sessionID = this.props.sessionID;
+    var candID = this.props.candID;
+
+    request = $.ajax({
+      type: "POST",
+      url: "ajax/thread_comment_bvl_feedback.php",
+      data: {"comment" : comment,
+      "feedbackID" : feedbackID,
+      "candID" : candID},
+      success: function (response) {
+        that.loadServerState();
+      },//end of success function
+      error: function (xhr, desc, err){
+        console.log(xhr);
+        console.log("Details: " + desc + "\nError:" + err);
+      }
+    })
+  },
+  render: function() {
+    var d = new Date();
+    var feedbackID = this.props.feedbackID;
+    if (this.state.thread_entries_toggled){
+      var arrow = 'glyphicon glyphicon-chevron-down glyphs';
+      var threadEntries = this.state.thread_entries_loaded.map(function(entry){
+        return <tr className="thread_entry"><td colSpan="100%">{entry.UserID} on {entry.TestDate} commented:
+        <br/> {entry.Comment}</td></tr>}
+      );
+
+    }
+    else{
+      var arrow = 'glyphicon glyphicon-chevron-right glyphs';
+    }
+
+    if (this.props.status == 'closed'){
+      var buttonText = 'closed';
+      var buttonClass = 'btn btn-success dropdown-toggle btn-sm' ;
+      var dropdown = <li><a onClick={this.props.onClickOpen}>Open</a></li>;
+      var commentButton = null;
+    }
+    else if (this.props.status == 'opened'){
+      var buttonText = 'opened'
+      var buttonClass = 'btn btn-danger dropdown-toggle btn-sm';
+      var dropdown = <li><a onClick={this.props.onClickClose}>Close</a></li>;
+      var commentButton = <span className="glyphicon glyphicon-pencil" onClick={this.props.commentToggle}></span>;
+    }
+    return(
+      <tbody>
+	<tr>
+	{this.props.fieldname? <td>{this.props.fieldname}<br/>{this.props.type}</td> : <td>{this.props.type}</td>}
+   
+      <td>{this.props.author} on:<br/>{this.props.date}</td>
+      <td>
+      <div className="btn-group">
+      <button name ="thread_button" type="button" className={buttonClass} data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+      {buttonText}
+      <span className="caret"></span>
+      </button>
+      <ul className="dropdown-menu">
+      {dropdown}
+      </ul>
+      </div>
+      <span className={arrow} onClick={this.toggle_entries}></span>
+      {commentButton}
+      </td>
+      </tr>
+      { this.props.commentToggled ?
+        <CommentEntryForm user={this.props.user} onCommentSend={this.new_thread_entry} toggleThisThread={this.toggle_entries.bind(this)}/>: null }
+        {threadEntries}
+        </tbody>
+      );
+    }
+  });
+
+  var CommentEntryForm = React.createClass({
+    getInitialState: function(){
+      return {
+        value: null
+      };
+    },
+    sendComment: function(){
+      this.props.onCommentSend(this.state.value);
+      this.setState({
+        value: "Comment added!"
+      })
+      this.props.toggleThisThread;
```

Comment:
I assume this is meant to call the function. The brackets are missing at the end.